### PR TITLE
--to=json argument to return JSON object of build (without file system writes)

### PR DESCRIPTION
**Pull request by zachleat**
And Eleventy method too for programmatic usage.

* [x] Support `ndjson` output too via https://twitter.com/emarticor/status/1357977044181688321

### CODE REVIEW

File: src/Eleventy.js [hunk 1]
```diff
@@ -759,7 +780,18 @@ Arguments:
...
+      let promise;
+      if (to === "fs") {
+        promise = this.writer.write();
+      } else if (to === "json") {
+        promise = this.writer.getJSON("json");
+      } else if (to === "ndjson") {
+        promise = this.writer.getJSON("ndjson");
+      } else {
+        throw new Error(
+          `Invalid argument for \`Eleventy->executeBuild(${to})\`, expected "json" or "fs".`
```
Comment (pdehaan) [inline src/Eleventy.js:792]: Q: Does this error msg need to add the new "ndjson" option?

Comment (zachleat) [reply src/Eleventy.js:792]: Yes thank you!

**Comment by pdehaan** [comment]
Either I'm not using this correctly, or I'm getting a bit of unwanted debugging data in my stdout.

```sh
# Install custom 11ty from your branch.
npm i github:11ty/eleventy#zachleat/no-fs-tojson -D
```

```sh
# Run locally installed eleventy w/ new `--to=json` flag and pipe output to "www.json".
npx eleventy --to=json > www.json
```

```js
Benchmark (Data): `./src/_data/github.js` took 189ms (33.3%)
[
  {"url":"/howto/2021-02-03-deploying-to-github-pages/","inputPath":"./src/howto/2021-02-03-deploying-to-github-pages/index.md","content":"<!DOCTYPE html>\n<html lang=\"en\">\n...</html>"},
  ...
]
```

The part that isn't quite right is that I still seem to get occasional debugging info being console'ed to stdout, which will break the JSON-ness of it:

> "Benchmark (Data): `./src/_data/github.js` took 189ms (33.3%)"

Similarly if I run <kbd>npx eleventy --to=ndjson > www.ndjson</kbd> I get something like the following on my dummy site:

```ndjson
{"url":"/","inputPath":"./src/index.md","content":"<!DOCTYPE html>\n<html lang=\"en\">...\n</html>\n"}
{"url":"/foo","inputPath":"./src/foo.md","content":"<!DOCTYPE html>\n<html lang=\"en\">...\n</html>\n"}
...
Benchmark (Data): `./src/_data/github.js` took 210ms (43.3%)
```

Initially I was running it through an <kbd>npm build</kbd> command for convenience, but that added even more stdout content from npm into the output file.

**Comment by zachleat** [comment]
Thanks for testing! Still working through this one :)

**Comment by zachleat** [comment]
Okay I think this is shippable now 🎉

**Comment by zachleat** [comment]
cc @jlengstorf This PR was inspired by our discussion last week!

Sample programmatic API script:

image

**Comment by zachleat** [comment]
Shipping

**Pull request merged**